src/writeback/rust.rs

```diff
@@ -0,0 +1,184 @@
+use super::WriteBackError;
+use crate::check::{ColumnChange, ModelChanges};
+use dbt_serde_yaml;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use std::path::Path;
+
+#[derive(Debug, Deserialize, Serialize)]
+struct ModelDoc {
+    name: Option<String>,
+    description: Option<String>,
+    columns: Vec<ColumnDoc>,
+    #[serde(flatten)]
+    extras: BTreeMap<String, dbt_serde_yaml::Value>,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+struct ColumnDoc {
```

Comment:
What if I instead, had every `Change` be related to the Doc of record, ModelChange->ModelDoc, and then just define the execute path for each variant?